src/datasets/arrow_dataset.py

```diff
@@ -2653,6 +2653,20 @@ def add_elasticsearch_index(
             )
         return self
 
+    def add_item(self, item: dict):
```

Comment:
I think we should make this method return a new Dataset object for consistency with all the other methods.
WDYT ?

Reply:
Right! Done @lhoestq.